### PR TITLE
Add longtime conformance test

### PR DESCRIPTION
**Pull request by edmond-zhu**
1. Modified video source to support loop mode.
2. Add longtime conformance test with timeout exit.

### CODE REVIEW

**Review by Cidana-Developers — COMMENTED** [review]
Regarding the long time test, maybe we can just provide a loop feature of video source to simulate the big file.

**Review by Cidana-Developers — APPROVED** [review]

**Review by Cidana-Developers — APPROVED** [review]

**Comment by Cidana-Developers** [comment]
Please attached the final checklist.

**Comment by edmond-zhu** [comment]
# For E2E longtime test:

# Pr for add longtime test for e2e conformance test case
## license: check if license is ok
## Comments
## warnings in Local build
1. mac: no new warning, only warning: -pie being ignored. It is only used when linking a main executable
2. linux: GNU warning in SpatialFullDistortionTest.cc
3. windows: none

## warnings on remote ci servers
1. jenkins:
a. clang-format
b. 
2. travis
  a. mac: `-pie being ignored. It is only used when linking a main executable`, can be ignored.
  b. linux: SpaticalFullDistoration wanring, could be fixed by merge the latest code;
3. appvedor
  a. no warnings

## dry run output
~~~
Running main() from C:\Works\AV1\SVT-AV1\third_party\googletest\src\gtest_main.cc
Note: Google Test filter = */LongtimeConformanceTest.*/*
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from SvtAv1/LongtimeConformanceTest
[ RUN      ] SvtAv1/LongtimeConformanceTest.DISABLED_LongtimeTest/default_setting
Start test case default_setting: test vector: kirland_640_480_30.yuv
Running main() from C:\Works\AV1\SVT-AV1\third_party\googletest\src\gtest_main.cc
Start test case default_setting: test vector: kirland_640_480_30.yuv
SVT [version]:  SVT-AV1 Encoder Lib v0.6.0
SVT [build]  :  Visual Studio 2017 64 bit
LIB Build date: Jul 19 2019 16:42:54
-------------------------------------------
Number of logical cores available: 12
Number of PPCS 72
-------------------------------------------
SVT [config]: Main Profile      Tier (auto)     Level (auto)
SVT [config]: EncoderMode                                                       : 8
SVT [config]: EncoderBitDepth / EncoderColorFormat / CompressedTenBitFormat     : 8 / 1 / 0
SVT [config]: SourceWidth / SourceHeight                                        : 640 / 480
SVT [config]: FrameRate / Gop Size                                              : 30 / 32
SVT [config]: HierarchicalLevels / BaseLayerSwitchMode / PredStructure          : 4 / 0 / 2
SVT [config]: BRC Mode / QP  / LookaheadDistance / SceneChange                  : CQP / 50 / 33 / 0
-------------------------------------------
SVT Memory Usage:
    total allocated memory:       1.61 GB
        malloced memory:          1.19 GB
        callocated memory:        36.60 MB
        allocated aligned memory: 396.30 MB
    mutex count: 573
    semaphore count: 345
    thread count: 75
    hash table fulless: 0.032791, hash bucket is healthy
top 10 malloced memory locations:
(227.86 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbPictureControlSet.c:1201
(144.96 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Encoder\Codec\EbEncHandle.c:3430
(143.28 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbModeDecisionProcess.c:91
(143.05 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbBitstreamUnit.c:40
(88.76 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbRestoration.c:1737
(43.59 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbCodingUnit.c:60
(37.64 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbModeDecisionProcess.c:92
(37.50 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbCodingUnit.c:61
(35.16 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbMotionEstimationContext.c:117
(35.16 MB): C:\Works\AV1\SVT-AV1\Source\Lib\Common\Codec\EbMotionEstimationContext.c:116
Encoder EOS
PSNR: 360 frames, total: 46.2110, luma: 43.6692, cb: 47.3095, cr: 47.6542
Enc Performance: 0.03sec/frame (32.5645FPS)
SVT: you have no memory leak
[       OK ] SvtAv1/LongtimeConformanceTest.DISABLED_LongtimeTest/default_setting (33685 ms)
[----------] 1 test from SvtAv1/LongtimeConformanceTest (33687 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (33691 ms total)
[  PASSED  ] 1 test.
~~~

**Comment by edmond-zhu** [comment]
@FochQu please verify it with following command:
~~~
SvtAv1E2ETests --gtest_filter=*/LongtimeConformanceTest.*/* --gtest_also_run_disabled_tests
~~~

File: test/vectors/longtime_comformance_test.cfg [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+# file_name Y4M? color_format w h bit_depth compressed_10bit_output start_frame frame_count
+#
+# Lines start with # will be ignored.
+# File must saved in same folder with config file.
+# color_format support 420, 420p10 in current version.
+# If 'compressed_10bit_output' set on with a 420p10 color format input file, test app will feed video frame into encoder with compressed 10 bit format.
+# start_frame and frame_count will set test range of a input file.
+# With 'frame_count = 0', will use all following frames start from 'start_frame'.
+# start_frame use 0 as first frame index.
+kirland_640_480_30.yuv 0 420 640 480 8 0 0 360
```
Comment (edmond-zhu) [inline test/vectors/longtime_comformance_test.cfg:12]: just a double check for the length of 360 frames seems a little bit short for a "longtime" test.

Comment (Cidana-Developers) [reply test/vectors/longtime_comformance_test.cfg:12]: Yes, I just want to test it.:)

**Review by FochQu — APPROVED** [review]
Branch for add_e2e_longtime_test
------------------------------------
 +Mac:
ld: warning: -pie being ignored. It is only used when linking a main executable

 +Linux
no warnings

 +Windows
no warnings

dry run output
--
test configuration file can not be opended: /home/jenkins/workspace/stream/longtime_comformance_test.cfg!
Note: Google Test filter = */LongtimeConformanceTest.*/*
[==========] Running 1 test from 1 test case.
[----------] Global test environment set-up.
[----------] 1 test from SvtAv1/LongtimeConformanceTest
[ RUN      ] SvtAv1/LongtimeConformanceTest.DISABLED_LongtimeTest/default_setting
[       OK ] SvtAv1/LongtimeConformanceTest.DISABLED_LongtimeTest/default_setting (0 ms)
[----------] 1 test from SvtAv1/LongtimeConformanceTest (0 ms total)

[----------] Global test environment tear-down
[==========] 1 test from 1 test case ran. (1 ms total)
[  PASSED  ] 1 test.

@FochQu 
no windows platform build info, and the dry-run does not match with original's dry-run, why approved?

**Pull request merged**